### PR TITLE
Dockerfile: Use the ubi-minimal image

### PR DESCRIPTION
**Pull request by raghavendra-talur**
This reduces the size of the ramen operator image by half. The new image size is 57% of the old image.

This PR also removes the GO111MODULE=on flag as that has been the
default for quite a few releases in Go.

```
quay.io/ramendr/ramen-operator               latest       551ac1797930  6 seconds ago   284 MB
quay.io/ramendr/ramen-operator               minimal      f1e115aa35ce  3 hours ago     164 MB
```

### CODE REVIEW

**Comment by raghavendra-talur** [comment]
> @raghavendra-talur don't we need ci.yaml and go.mod updated as well?

The go version update was hanlded in a different PR now. Now this PR only reduces the image size.

File: Dockerfile [hunk 1]
```diff
@@ -18,9 +18,9 @@ COPY main.go main.go
 COPY controllers/ controllers/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
```
Comment (nirs) [inline Dockerfile:21]: Is this change related? If we need to revert the image change, it will also revert this change. If you split it to 2 commits we don't have this issue.

Comment (raghavendra-talur) [reply Dockerfile:21]: Done

**Review by nirs — APPROVED** [review]
